### PR TITLE
Add timezone config to mysqlDaoQueryHelper and tests.

### PR DESCRIPTION
**Pull request by cdeveas**

### CODE REVIEW

File: lib/mysqlService.js [hunk 1]
```diff
@@ -198,7 +202,7 @@ class MysqlService {
     }
 
     clean(input) {
-        return Mysql.escape(input);
+        return Mysql.escape(input, false, this._writeConnectionConfig.timezone);
```
Comment (cdeveas) [inline lib/mysqlService.js]: This now defaults the queryHelper to the timezone specified in the writeConfig

File: lib/mysqlDaoQueryHelper.js [hunk 2]
```diff
@@ -3,6 +3,7 @@
 const _ = require('lodash');
 const Squel = require('squel');
 const VO = require('hapiest-vo');
+const moment = require('moment');
```
Comment (jonathan-fulton) [inline lib/mysqlDaoQueryHelper.js]: Capitalize for consistency.

File: lib/mysqlDaoQueryHelper.js [hunk 3]
```diff
@@ -239,8 +240,7 @@ class MysqlDaoQueryHelper {
                 // Escaping primitive values
                 const cleanValue = config.dontCleanMysqlFunctions ? this.cleanSpecial(uncleanValue) : this.clean(uncleanValue);
                 cleanValues[snakeCaseProperty] = cleanValue;
-            } else if (_.isDate(uncleanValue)) {
-                //TODO: Handle timezones using a config setting?
+            } else if (_.isDate(uncleanValue) || uncleanValue instanceof moment) {
```
Comment (jonathan-fulton) [inline lib/mysqlDaoQueryHelper.js]: Update capitalization to agree with above change.

File: lib/mysqlService.js [hunk 4]
```diff
@@ -16,6 +16,10 @@ class MysqlService {
      * @param {Logger} [logger] - instance of hapiest-logger
      */
     constructor(writeConnectionConfig, readConnectionConfig, logger) {
+
+        this._writeConnectionConfig = writeConnectionConfig;
```
Comment (jonathan-fulton) [inline lib/mysqlService.js]: Do we need the entire configs to be saved on the instance?  If not, let's just save off the variable(s) we need, which from what I can tell looks like timezone.

File: lib/mysqlService.js (hunk 1 above)
Comment (jonathan-fulton) [inline lib/mysqlService.js]: Config.timezone isn't necessarily defined. See mysqlPoolConnectionConfigSchema/Factory.  Feel free to get mad at me for no documentation as well :)

I'd say do a simple if/else checking existence of it.

File: lib/mysqlDaoQueryHelper.js (hunk 2 above)
Comment (jonathan-fulton) [inline lib/mysqlDaoQueryHelper.js]: Capitalize for consistency.

File: lib/mysqlDaoQueryHelper.js (hunk 3 above)
Comment (jonathan-fulton) [inline lib/mysqlDaoQueryHelper.js]: Capitalize per above.

File: lib/mysqlService.js (hunk 4 above)
Comment (jonathan-fulton) [inline lib/mysqlService.js]: Do we need the full configs saved?  If not, let's just save what we need, which appears to be the timezone setting.

File: test/unit/test-mysqlDaoQueryHelper.js [hunk 5]
```diff
@@ -152,13 +183,13 @@ describe('MysqlDaoQueryHelper', function() {
         it('Should generate a SELECT statement with a query object and gt/lt', function() {
             const sql = mysqlDaoQueryHelper.getAll({
                 dateAdded: {
-                    gt: new Date('1990-01-05T13:30:00Z'),
+                    gte: new Date('1990-01-05T13:30:00Z'),
```
Comment (jonathan-fulton) [inline test/unit/test-mysqlDaoQueryHelper.js]: Is this related to the timezone stuff?

File: test/unit/test-mysqlDaoQueryHelper.js [hunk 6]
```diff
@@ -103,6 +104,36 @@ describe('MysqlDaoQueryHelper', function() {
             sql.should.eql("SELECT * FROM users WHERE (first_name IS NULL) AND (last_name = 'Doe') LIMIT 1");
         });
 
+        it('Should generate dates with for a utc timezone ', function() {
```
Comment (jonathan-fulton) [inline test/unit/test-mysqlDaoQueryHelper.js]: I would add test cases in the "clean" section below in this file.  Also, I'd put the before/after stuff at the top of the describe block.

File: test/unit/test-mysqlDaoQueryHelper.js [hunk 7]
```diff
@@ -103,6 +104,36 @@ describe('MysqlDaoQueryHelper', function() {
...
+            const utcDateString = now.format('YYYY-MM-DD HH:mm:ss.SSS');
+
+            const sql = mysqlDaoQueryHelper.getOne({dateAdded: nowDate});
+            Should.exist(sql);
+
+            sql.should.eql(`SELECT * FROM users WHERE (date_added = '${utcDateString}') LIMIT 1`);
+        });
+
+        before(() => {
+            mysqlDaoQueryHelper._clean = value => Mysql.escape(value);
```
Comment (jonathan-fulton) [inline test/unit/test-mysqlDaoQueryHelper.js]: Instead of before/after, I would probably create two different helpers, one with a local timezone clean function and one with a UTC clean function.  Should be straight forward b/c the clean function is passed in via the constructor.

File: test/unit/test-mysqlDaoQueryHelper.js (hunk 6 above)
Comment (cdeveas) [inline test/unit/test-mysqlDaoQueryHelper.js]: Done

File: test/unit/test-mysqlDaoQueryHelper.js (hunk 5 above)
Comment (cdeveas) [inline test/unit/test-mysqlDaoQueryHelper.js]: Nope just me messing
